spiceConcurrentCall_test.py

```diff
@@ -0,0 +1,109 @@
+# %%
+import spiceypy as spice # this imports the version 4.0.0
+import os
+
+# %%
+import functools
+import threading
+
+from spiceypy.spiceypy import spkezr
+
+_spicelock = threading.RLock()
+
+enablethreadinglock = False
+
+def spicelock_for_multithread(f):
+    """
+    Decorator for spiceypy to avoid concurrent calls to cspice lib.
+
+    :return:
+    """
+    @functools.wraps(f)
+    def lock(*args, **kwargs):
+        if enablethreadinglock:
+            with _spicelock:
+                try:
+                    res = f(*args, **kwargs)
+                    return res
+                except BaseException:
+                    raise
+        else:
+            try:
+                res = f(*args, **kwargs)
+                return res
+            except BaseException:
+                raise
+
+    return lock
+
+# %%
+from multiprocessing.dummy import Pool as ThreadPool
+
+
+
+# %%
+b1900 = spice.b1900
+tb1900 = spicelock_for_multithread(b1900)
+
+# %%
+print('Test thread pool without locking for b1900')
+spice.enablethreadinglock = False
+with ThreadPool() as pool:
+    test_b1900 = pool.map(lambda x: tb1900(), range(10000))
+
+print('Test thread pool with locking for b1900')
+spice.enablethreadinglock = True
+with ThreadPool() as pool:
+    test_tb1900 = pool.map(lambda x: tb1900(), range(10000))
+
+
+
+# %%
+pathKernels = 'path/to/kernels'
```

Comment:
See example test functions in move to a new test function within https://github.com/AndrewAnnex/SpiceyPy/blob/main/spiceypy/tests/test_wrapper.py for this, I have other existing tests for spkezr and my own code for managing loading test kernels for the test suite so some of the code here can be simplified by following those examples as models

Reply:
As said before, I am not sure about the new test function. Please check it for me :)